### PR TITLE
Basic functionality, one thread for every client, consistent calculate factorial

### PR DESCRIPTION
**Pull request by khrushv**

### CODE REVIEW

File: src/server/CMakeLists.txt [hunk 1]
```diff
@@ -6,4 +6,11 @@ add_executable(server server.c opt_arg.c)
 find_package(Threads REQUIRED)
 target_link_libraries(server ${CMAKE_THREAD_LIBS_INIT})
 
-target_link_libraries(server thread_pool)
\ No newline at end of file
+
+target_include_directories(server PUBLIC ../utils)
```
Comment (sergeyampo) [inline src/server/CMakeLists.txt]: There's no utils folder

File: src/server/CMakeLists.txt [hunk 2]
```diff
@@ -6,4 +6,11 @@ add_executable(server server.c opt_arg.c)
...
+
+
+target_link_libraries(server )
```
Comment (sergeyampo) [inline src/server/CMakeLists.txt]: ?

File: src/server/server.c [hunk 3]
```diff
@@ -16,178 +16,166 @@
...
-  return ans;
+    return ans;
 }
 
 void *ThreadFactorial(void *args) {
-  struct FactorialArgs *fargs = (struct FactorialArgs *) args;
-  return (void *) (uint64_t *) Factorial(fargs);
+    struct FactorialArgs *fargs = (struct FactorialArgs *) args;
+    return (void *) (uint64_t *) Factorial(fargs);
 }
```
Comment (sergeyampo) [inline src/server/server.c]: Why do we need this?

File: src/server/server.c [hunk 4]
```diff
@@ -16,178 +16,166 @@
...
-    socklen_t client_len = sizeof(client);
-    int client_fd = accept(server_fd, (struct sockaddr *) &client, &client_len);
-
-    if (client_fd < 0) {
-      fprintf(stderr, "Could not establish new connection\n");
-      continue;
-    }
 
+
+uint64_t naive_factorial(uint64_t n)
```
Comment (sergeyampo) [inline src/server/server.c]: naive?

**Review by sergeyampo — CHANGES_REQUESTED** [review]

File: src/server/server.c (hunk 4 above)
Comment (sergeyampo) [reply src/server/server.c]: uint64_t n => const uint64_t* n

File: src/server/CMakeLists.txt (hunk 1 above)
Comment (khrushv) [reply src/server/CMakeLists.txt]: I'll remove one

File: src/server/CMakeLists.txt (hunk 2 above)
Comment (khrushv) [reply src/server/CMakeLists.txt]: I'll remove it

File: src/server/server.c (hunk 3 above)
Comment (khrushv) [reply src/server/server.c]: We don't need it, so I'll delete it

File: src/server/server.c (hunk 4 above)
Comment (khrushv) [reply src/server/server.c]: I will rename the function and change the parameter

**Pull request closed without merging**